casper/src/main/resources/RevVault.rho

```diff
@@ -175,100 +219,139 @@ in {
                     }
+                  } else {
+                    // Unsuccessful transfers are not logged
+                    ret!(result)
                   }
                 }
-              } |
-              // Set to Nil to disable logging
-              contract revVault(@"setLog", logCh, authKey, ret) = {
-                new authKeyValidCh in {
-                  @AuthKey!("check", *authKey, (*_revVault, ownRevAddress), *authKeyValidCh) |
-                  for (@result <- authKeyValidCh) {
-                    if (result) {
-                      new ack in {
-                        for (_ <- logStore) {
-                          logStore!(*logCh) |
-                          // Set logging on the purse so that we also see deposits into the vault
-                          purse!("setLog", *logCh, *ack) |
-                          for (_ <- ack) {
-                            ret!(true)
-                          }
-                        }
-                      }
-                    } else {
-                      ret!(false)
-                    }
+              }
+            } |
+            // Returns an entry of the vault's log
+            contract revVault(@"readLogEntry", ret) = {
+              new thmRet in {
+                // Lookup the log channel associated with `fromRevAddress`
+                TreeHashMap!("get", vaultMap, fromRevAddress, *thmRet) |
+                for (@(_, _, *logCh) <- thmRet) {
+                  // Read a transfer entry and return it on `ret`
+                  for (@("transfer", entry) <<- logCh) {
+                    ret!(entry)
                   }
                 }
-              } |
-              // Internal RevVault method for making deposits between purses.
-              contract @{ownRevAddress | bundle0{*_revVault}}(@"_deposit", depositPurse, retCh) = {
-                _depositTemplate!(*purse, *depositPurse, *retCh)
               }
+            } |
+            // Internal RevVault method for making a deposit from purse in `_revVault`
+            contract @{fromRevAddress | bundle0{*_revVault}}(@"_deposit", depositPurse, retCh) = {
+              _depositTemplate!(*purse, *depositPurse, *retCh)
             }
           } |
           // Checks all necessary validation and authentication prerequisites. If everything checks out,
-          // amount is transferred from purse (ownRevAddress) to vault at targetAddress.
-          contract _transferTemplate(@ownRevAddress, purse, @targetAddress, @amount, authKey, ret) = {
-            new revAddressValid, revAddressValidEither, amountNonNegative,
-                authKeyValidCh, authKeyValidEitherCh,
-                parametersOkCh, parametersAndAuthOkCh,
-                split, eitherPurseCh, doDeposit
+          // `amount` is transferred from vault at `fromRevAddress` to vault at `toRevAddress`.
+          contract _transferTemplate(@fromRevAddress, purse, @toRevAddress, @amount, authKey, ret) = {
```

Comment:
another naming conflicts `fromRevAddress` and `toRevAddress` compared to `ownerRevAddress` and `targetRevAddress`